client/service.go

```diff
@@ -455,7 +455,12 @@ func (cm *ConnectionManager) realConnect() (net.Conn, error) {
 	protocol := cm.cfg.Protocol
 	if protocol == "websocket" {
 		protocol = "tcp"
-		dialOptions = append(dialOptions, libdial.WithAfterHook(libdial.AfterHook{Hook: frpNet.DialHookWebsocket()}))
+		dialOptions = append(dialOptions, libdial.WithAfterHook(libdial.AfterHook{
+			// make sure to do the websocket handshake after the TLS handshake,
+			// to support websocket over TLS.
+			Priority: libdial.WithTLSConfigPriority + 1,
```

Comment:
This change has the potential to break the e2e case located at https://github.com/fatedier/frp/blob/dev/test/e2e/basic/client_server.go#L100. 

We would like to avoid making any incompatible changes.

In my opinion, `tls over websocket` is not equivalent to `wss`. 

Therefore, if we intend to support wss, it would be better to move it from the protocol to the proxy parameter.

Reply:
oh okay! I didn't realize "TLS over websocket" was actually a desired combination. 

Can you clarify which exact `frpc.ini` settings should result in "websocket over TLS"? I'll update the code to match that. Thanks!

Reply:
Perhaps an alternative solution could be to add support for `wss` to the `protocol` parameter. 
If the user configuration is set to `websocket`, it will remain unchanged. However, if the configuration is set to `wss`, the websocket and tls order can be adjusted to implement wss.

WDYT?